### PR TITLE
Apply ignore patterns

### LINKED ISSUE #66: Ignore option doesn't work.
Hello,

I have the following settings :
wpbhk50

But it doesn't work :
screenshot_20

### PR DESCRIPTION
**Pull request by skyrpex**
Fixes #66.

### CODE REVIEW

File: lib/main.coffee [hunk 1]
```diff
@@ -64,6 +63,11 @@ module.exports =
       lintOnFly: true
       lint: (textEditor) =>
         filePath = textEditor.getPath()
+
+        # Check if file should be ignored
+        baseName = path.basename filePath
+        return [] if _.any @ignore, (pattern) => minimatch baseName, pattern
```
Comment (Arcanemagus) [inline lib/main.coffee]: Shouldn't this be `->` instead of `=>`?

**Comment by Arcanemagus** [comment]
Ha, for a second I thought this had completely broken the package till I remembered I had put `*.php` in the Ignore field to test earlier :stuck_out_tongue:

Looks good to me, other than the comments above, as soon as those are fixed I'll merge it in.

File: package.json [hunk 2]
```diff
@@ -10,7 +10,9 @@
   },
   "dependencies": {
     "atom-linter": "^3.0.0",
-    "atom-package-deps": "^2.1.3"
+    "atom-package-deps": "^2.1.3",
+    "lodash": "^3.10.1",
```
Comment (steelbrain) [inline package.json]: A whole dependency just to use `_.any`? Why not use `Array.prototype.some`?

**Comment by Arcanemagus** [comment]
This doesn't cause any issues as is, since we haven't gotten a response from @skyrpex I'm going to merge this and fix the style issues.

**Pull request merged**